src/hotspot/share/runtime/globals.hpp

```diff
@@ -2088,10 +2088,13 @@ const intx ObjectAlignmentInBytes = 8;
   develop(bool, TraceOptimizedUpcallStubs, false,                           \
                 "Trace optimized upcall stub generation")                   \
                                                                             \
-  product(ccstr, CRaCCheckpointTo, NULL, "Path to checkpoint image")        \
+  product(ccstr, CRaCCheckpointTo, NULL, MANAGEABLE,                        \
+        "Path to checkpoint image directory")                               \
                                                                             \
-  product(ccstr, CRaCRestoreFrom, NULL, "Path to image for restore, "       \
-      "replaces the initializing VM on success")                            \
+  /* It does not make sense to change this flag in runtime but we'll tag */ \
+  /* it MANAGEABLE to prevent warnings when setting this on restore. */     \
+  product(ccstr, CRaCRestoreFrom, NULL, MANAGEABLE,                         \
```

Comment:
Below we see that `CRaCIgnoredFileDescriptors` is RESTORABLE (I would rather use SET_ON_RESTORE as all flags are restored at its previous value) but not MANAGEABLE, so that's not perfect either. The reason why I rather stayed on MANAGEABLE was to prevent changing every `MANAGEABLE` to `MANAGEABLE | SET_ON_RESTORE`, that would complicate any backport from mainline.

I can do the SET_ON_RESTORE (as superset of MANAGEABLE) but I don't think that the few exceptions that could be rather documented by a few lines of comment really deserve a separate flag.

Reply:
> The reason why I rather stayed on MANAGEABLE was to prevent changing every `MANAGEABLE` to `MANAGEABLE | SET_ON_RESTORE`, that would complicate any backport from mainline.

I tottaly support MANAGEABLE => RESTORABLE, as it makes sense.

> I can do the SET_ON_RESTORE (as superset of MANAGEABLE) but I don't think that the few exceptions that could be rather documented by a few lines of comment really deserve a separate flag.

Option are also some form of the documentation for the flags. So the new class of options deserve their own name.

> Below we see that `CRaCIgnoredFileDescriptors` is RESTORABLE but not MANAGEABLE, so that's not perfect either. 

Yes, that is another one. And there are many more, like PrintCompilation. Or any other product flag which handling in the VM allows update on restore. Over time the set of RESTORABLE flags can grow as VM implementation allows. While there is a higher bar to include a flag into MANGEABLE set [1]

[1] https://github.com/openjdk/crac/blob/crac/src/hotspot/share/runtime/globals.hpp#L86

> (I would rather use SET_ON_RESTORE as all flags are restored at its previous value)

I don't like RESTORABLE as well (relates to RESTORE, but how is not clear). The new name better to fit into existing set of: DIAGNOSTIC, EXPERIMENTAL, or MANAGEABLE.